domain/schema/model.go

```diff
@@ -158,6 +159,32 @@ INSERT INTO change_log_namespace VALUES
 `)
 }
 
+func modelSchema() schema.Patch {
+	return schema.MakePatch(`
+-- The model table represents a readonly denormalised model data. The intended
+-- use is to provide a read-only view of the model data for the purpose of
+-- accessing common model data without the need to span multiple databases.
+CREATE TABLE model (
+    uuid            TEXT PRIMARY KEY,
+    name            TEXT NOT NULL,
+    type            TEXT NOT NULL,
+    cloud           TEXT NOT NULL,
+    cloud_region    TEXT NOT NULL
+);
+
+-- A unique constraint over a constant index ensures only 1 entry matching the
+-- condition can exist.
+CREATE UNIQUE INDEX idx_singleton_model ON model ((1));
+
+-- The model table is read-only, so we create a trigger to prevent updates.
+CREATE TRIGGER trg_readonly_model
+BEFORE UPDATE ON model
```

Comment:
What about delete?

Reply:
I can add it, but I wanted an escape hatch to be able to delete the model, otherwise you've bricked your model (apart from deleting the triggers, which would allow that to happen :thinking: )